R/try_query.R

```diff
@@ -34,12 +38,18 @@ try_query <-
 
     query_result <- "empty"
     try_n <- 0
+    # Note: The handling of retries in case of a timeout could be adjusted in the future.
+    # For now I introduced try_if_timeout, but potentially it makes sense to always retry even
+    # in case of a timeout. Then however, the number of retries needs to be adjusted for some
+    # functions that retrieve a lot of data.
     while (!is(query_result, "response") &
       try_n < max_tries &
+      # this ifelse stops requery if they timeout except for if try_if_timeout is TRUE
       !ifelse(is(query_result, "character"),
-        stringr::str_detect(query_result, pattern = "Timeout was reached"),
+        stringr::str_detect(query_result, pattern = "Timeout was reached") & !try_if_timeout,
         FALSE
-      )) { # this ifelse stops requery if the timeout is too low.
+      )
+      ) {
```

Comment:
i would add more verbous error messages like https://github.com/jpquast/protti/blob/fetch_uniprot_252/R/try_query.R line 59-70

Reply:
added this.